ext/bcmath/libbcmath/src/recmul.c

```diff
@@ -38,23 +38,26 @@
 
 
 #if SIZEOF_SIZE_T >= 8
-#  define BC_MUL_UINT_DIGITS 8
-#  define BC_MUL_UINT_OVERFLOW (BC_UINT_T) 100000000
+#  define BC_VECTORIZED_MULTIPLICATION_VECTOR_SIZE 8
+/* The boundary number is computed from BC_VECTORIZED_MULTIPLICATION_VECTOR_SIZE times BASE */
+#  define BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER (BC_INT_T) 100000000
+#  define BC_MUL_MAX_ADD_COUNT (((BC_INT_T) 0x0fffffffffffffff) / \
+		(BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER * BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER))
 #else
-#  define BC_MUL_UINT_DIGITS 4
-#  define BC_MUL_UINT_OVERFLOW (BC_UINT_T) 10000
+#  define BC_VECTORIZED_MULTIPLICATION_VECTOR_SIZE 4
+#  define BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER (BC_INT_T) 10000
+#  define BC_MUL_MAX_ADD_COUNT (((BC_INT_T) 0x0fffffff) / \
+		(BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER * BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER))
 #endif
 
-#define BC_MUL_MAX_ADD_COUNT (~((BC_UINT_T) 0) / (BC_MUL_UINT_OVERFLOW * BC_MUL_UINT_OVERFLOW))
-
 
 /* Multiply utility routines */
 
-static inline void bc_digits_adjustment(BC_UINT_T *prod_uint, size_t prod_arr_size)
+static inline void bc_digits_adjustment(BC_INT_T *prod_int, size_t prod_arr_size)
 {
 	for (size_t i = 0; i < prod_arr_size - 1; i++) {
-		prod_uint[i + 1] += prod_uint[i] / BC_MUL_UINT_OVERFLOW;
-		prod_uint[i] %= BC_MUL_UINT_OVERFLOW;
+		prod_int[i + 1] += prod_int[i] / BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER;
+		prod_int[i] %= BC_VECTORIZED_MULTIPLICATION_BOUNDARY_NUMBER;
```

Comment:
Okay now I understand what this function actually does, wouldn't a better name for the function be something like `bc_split_number()` as it seems to be splitting the number into a high and low part?

Reply:
Thx, fixed it